docs/proposals/initializers.md

```diff
@@ -0,0 +1,110 @@
+
+**ObjectMeta.Initialized**
+
+Boolean flag denoting if resource is or is not initialized. If a resource has active initializers, the value is false. Once all initializers have acted, the value is toggled to true. Once a resource is initialized, it is forever initialized. When a resource is not initialized, validation rules for editing are loosened to allow components to make delayed updates. An object must pass strict validation in order to become "initialized".
+
+## REST API Changes
+
+All API objects will support a new endpoint as a sub-resource that allows non-initialized objects to be edited via a PUT that does not carry the same validation constraints as normal updates of a resource.
+
+For example:
+
+```PUT /api/{version}/{group}/{kind}/{name}/initialize/{initializer-name}```
+
+The input to the request would be the normal serialization of the object.  The input object would be persisted, and the named *initializer* would be removed from the object's set of **ObjectMeta.Initializers**.  The only validation that must pass on the object is the validation of the *ObjectMeta* related data.  If the named object was already initialized, the API server would error.  If the initializer was not in the list of **ObjectMeta.Initializers**, the request would error.
+
+In addition, each object would support an API to mark the object as completely initialized.
+
+For example:
+
+```POST /api/initialize```
+
+The input to the request would be a *Initialize* object, that would work similar to *Scale* with HPA that would take a subresource reference to know the object that is to be transitioned from non-initialized to initialized.  If the object was already initialized,
+the API would error.  If the object failed strict validation, the system would error.
+
+**NOTE: I will go back and fill in more details here**
+
+## Making the transition to initialized
+
+A new controller is introduced into the system that looks for objects that have no more named "initializers" but are not yet "initialized".
+
+It introspects the set of registered *Initializers*, and uses the *Discovery API* and a generic API client to know what resource types to WATCH that are not yet initialized.  It then creates a set of watchers that look for objects whose list of "initializers" is empty, but the object is not yet "initialized".
+
+It then performs an "/api/initialize" action for each resource that meets the required criteria.  If the API action succeeded, the object is now initialized for use by the system.  If the API action fails, it's considered to be in an error state that is unrecoverable.  When this occurs, the controller will permanently delete the resource from the API server and log an event because the object would forever fail strict validation.
```

Comment:
@thockin - is time the factor here? if an initializer is errant for pods, its possible it would prevent all pods from ever being created.  basically, this was a catch-all for what to do if the initialized object does not pass strict validation.  seems like information in the event listing is a fine place to start debugging?